geomesa-plugin/src/main/scala/org/locationtech/geomesa/plugin/wcs/GeoMesaCoverageReader.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2014 Commonwealth Computer Research, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.locationtech.geomesa.plugin.wcs
+
+import com.typesafe.scalalogging.slf4j.Logging
+import org.geotools.coverage.CoverageFactoryFinder
+import org.geotools.coverage.grid.io.AbstractGridCoverage2DReader
+import org.geotools.coverage.grid.{GridCoverage2D, GridEnvelope2D}
+import org.geotools.factory.Hints
+import org.geotools.geometry.GeneralEnvelope
+import org.geotools.referencing.CRS
+import org.geotools.referencing.crs.DefaultGeographicCRS
+import org.geotools.util.Utilities
+import org.joda.time.format.DateTimeFormat
+import org.joda.time.{DateTime, DateTimeZone}
+import org.locationtech.geomesa.raster.data.{AccumuloRasterStore, GeoMesaCoverageQueryParams}
+import org.opengis.parameter.GeneralParameterValue
+
+import scala.util.Try
+
+object GeoMesaCoverageReader {
+  val GeoServerDateFormat = DateTimeFormat.forPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
+  val DefaultDateString = GeoServerDateFormat.print(new DateTime(DateTimeZone.forID("UTC")))
+  val FORMAT = """accumulo://(.*):(.*)@(.*)/(.*)#zookeepers=([^#]*)(?:#auths=)?([^#]*)(?:#visibilities=)?([^#]*)(?:#collectStats=)?([^#]*)$""".r
+}
+
+import org.locationtech.geomesa.plugin.wcs.GeoMesaCoverageReader._
+
+class GeoMesaCoverageReader(val url: String, hints: Hints) extends AbstractGridCoverage2DReader() with Logging {
+  logger.debug(s"""creating coverage reader for url "${url.replaceAll(":.*@", ":********@").replaceAll("#auths=.*","#auths=********")}"""")
+  val FORMAT(user, password, instanceId, table, zookeepers, auths, visibilities, collectStats) = url
+  logger.debug(s"extracted user $user, password ********, instance id $instanceId, table $table, zookeepers $zookeepers, auths ********")
+  logger.info(s"collectstats: $collectStats")
+  coverageName = table
+
+  val dostats = if (collectStats == "null") false else true
+  val ars = AccumuloRasterStore(user, password, instanceId, zookeepers, table, auths, visibilities, collectStats = dostats)
+
+  // TODO: Either this is needed for rasterToCoverages or remove it.
```

Comment:
I would suggest either removing this comment (iff we know the resolution), or create a new ticket for the known issue.